### PR TITLE
adding getKeyPaths code

### PR DESCRIPTION
**Pull request by moonpatroller**
## Details
**Issue you fixed:** https://github.com/rgehan/hacktoberfest-2k18-katas/issues/680

**Issue you added:** https://github.com/rgehan/hacktoberfest-2k18-katas/issues/683

**Name of the function you implemented**: getKeyPaths

**Name of the function you added test cases for**: createHeap

## Checklist (won't be merged if they are not all ticked)
- [x] I have asked for this issue, and was assigned it
- [x] I have properly filled the "Details" section above  :arrow_up:
- [x] I didn't modify any test case  :red_circle:
- [x] I have implemented the function and all its tests pass  :white_check_mark:
- [x] I have added a new test case for someone to implement  :new:
- [x] I have created an issue for my new test case  :speak_no_evil:
- [x] I've :star: the repository (or not, all up to you)
- [x] And more importantly, I've had fun! :beer:

### CODE REVIEW

**Comment by rgehan** [comment]
Hey @moonpatroller,

Thanks for your contribution, your PR was merged, and your kata is now available for others to play with.

If there's been any issue with the process, please ping me in your reply.

##### This comment was automatically generated by a bot.

File: src/getKeyPaths.js [hunk 1]
```diff
@@ -1,3 +1,17 @@
-export const getKeyPaths = () => {
-  // TO IMPLEMENT IN ANOTHER PR
+function getPaths(obj) {
+    const acc = [];
+    Object.entries(obj).map(([key, value]) => {
+        if (value === null || typeof value !== 'object') {
+            acc.push(key);
+        } else {
+            const subPaths = getPaths(value);
+            subPaths.map(p => key + '.' + p).forEach(p => acc.push(p));
```
Comment (sadarshannaiynar) [inline src/getKeyPaths.js]: This could be combined to do in the forEach itself. :)

File: src/getKeyPaths.js [hunk 2]
```diff
@@ -1,3 +1,17 @@
-export const getKeyPaths = () => {
-  // TO IMPLEMENT IN ANOTHER PR
+function getPaths(obj) {
+    const acc = [];
+    Object.entries(obj).map(([key, value]) => {
```
Comment (sadarshannaiynar) [inline src/getKeyPaths.js]: No need of map as it will create a array in the memory. Use `forEach` if you don't want a new array. :)

**Pull request merged**